index.js

```diff
@@ -10,50 +11,36 @@ export const ACCESSIBLE = {
   AFTER_FIRST_UNLOCK_THIS_DEVICE_ONLY:
     'AccessibleAfterFirstUnlockThisDeviceOnly',
   ALWAYS_THIS_DEVICE_ONLY: 'AccessibleAlwaysThisDeviceOnly',
-};
+});
 
-export const ACCESS_CONTROL = {
+export const ACCESS_CONTROL = Object.freeze({
   USER_PRESENCE: 'UserPresence',
   BIOMETRY_ANY: 'BiometryAny',
   BIOMETRY_CURRENT_SET: 'BiometryCurrentSet',
   DEVICE_PASSCODE: 'DevicePasscode',
   APPLICATION_PASSWORD: 'ApplicationPassword',
   BIOMETRY_ANY_OR_DEVICE_PASSCODE: 'BiometryAnyOrDevicePasscode',
   BIOMETRY_CURRENT_SET_OR_DEVICE_PASSCODE: 'BiometryCurrentSetOrDevicePasscode',
-};
+});
 
-export const AUTHENTICATION_TYPE = {
+export const AUTHENTICATION_TYPE = Object.freeze({
   DEVICE_PASSCODE_OR_BIOMETRICS: 'AuthenticationWithBiometricsDevicePasscode',
   BIOMETRICS: 'AuthenticationWithBiometrics',
-};
+});
 
-export const BIOMETRY_TYPE = {
+export const BIOMETRY_TYPE = Object.freeze({
   TOUCH_ID: 'TouchID',
   FACE_ID: 'FaceID',
   FINGERPRINT: 'Fingerprint',
-};
+});
+
+export type SecAccessible = $Values<typeof ACCESSIBLE>;
```

Comment:
`$Values<typeof ACCESSIBLE>` is the same as saying `string`. The accurate version would be `export type SecAccessible = 'AccessibleWhenUnlocked' | 'AccessibleAfterFirstUnlock' ... | 'AccessibleAlwaysThisDeviceOnly'`. Not sure what the best way to go about this is... Any suggestions?

Reply:
You are partly right, but `$Values`  works perfectly with frozen objects:

![image](https://user-images.githubusercontent.com/4661784/50207335-32d41300-037f-11e9-8ecb-131a8d564a70.png)

![image](https://user-images.githubusercontent.com/4661784/50207346-38c9f400-037f-11e9-8be0-fdeafe920105.png)

Reply:
interesting, I didn't know that. I guess that solves this.